doc/source/rllib/rllib-learner.rst

```diff
@@ -0,0 +1,459 @@
+.. include:: /_includes/rllib/announcement.rst
+
+.. include:: /_includes/rllib/we_are_hiring.rst
+
+.. |tensorflow| image:: images/tensorflow.png
+    :class: inline-figure
+    :width: 16
+
+.. |pytorch| image:: images/pytorch.png
+    :class: inline-figure
+    :width: 16
+
+
+Learner (Alpha)
+===============
+
+:py:class:`~ray.rllib.core.learner.learner.Learner` allows you to abstract the training 
+logic of RLModules. It supports both gradient-based and non-gradient-based updates (e.g.
+polyak averaging, etc.) The API enables you to distribute the Learner using data-
+distributed parallel (DDP). The Learner achieves the following:
+
+
+(1) Facilitates gradient-based updates on `RLModule`.
+(2) Provides abstractions for non-gradient based updates such as polyak averaging, etc.
+(3) Reporting training statistics.
+(4) Checkpoints the modules and optimizer states for durable training.
+
+The :py:class:`~ray.rllib.core.learner.learner.Learner` class supports data-distributed-
+parallel style training using the 
+:py:class:`~ray.rllib.core.learner.learner_group.LearnerGroup` API. Under this paradigm, 
+the :py:class:`~ray.rllib.core.learner.learner_group.LearnerGroup` maintains multiple 
+copies of the same :py:class:`~ray.rllib.core.learner.learner.Learner` with identical 
+parameters and hyperparameters. Each of these
+:py:class:`~ray.rllib.core.learner.learner.Learner` instances computes the loss and gradients on a
+shard of a sample batch and then accumulates the gradients across the 
+:py:class:`~ray.rllib.core.learner.learner.Learner` instances. Learn more about data-distributed 
+parallel learning in 
+`this article. <https://pytorch.org/tutorials/intermediate/ddp_tutorial.html>`_
+
+:py:class:`~ray.rllib.core.learner.learner_group.LearnerGroup` also allows for 
+asynchronous training and (distributed) checkpointing for durability during training.
+
+Enabling Learner API in RLlib experiments
+=========================================
+
+Adjust the amount of resources for training using the 
+`num_gpus_per_learner_worker`, `num_cpus_per_learner_worker`, and `num_learner_workers`
+arguments in the `AlgorithmConfig`.
+
+.. testcode::
+	:hide:
+    :skipif: True
+
+    from ray.rllib.algorithms.ppo.ppo import PPOConfig
```

Comment:
This doc contains a lot of code that is not tested.
There exists a folder "doc/source/rllib/doc_code" that could hold a guide file to hold all this test code that actually gets executed there. You could put code there and grab code snippets from there to ensure these example snippets stay up to date.

Reply:
I think lets do that in a later iteration at this point since doc picks are due today. 

But I agree that if thats the workaround then we'll go that way instead.